src/commands/list-kinesis-streams.js

```diff
@@ -0,0 +1,227 @@
+const { Command, flags } = require("@oclif/command");
+const { checkVersion } = require("../lib/version-check");
+const Lambda = require("../lib/lambda");
+require("colors");
+const uuid = require("uuid");
+
+const ONE_MIN_IN_SECONDS = 60;
+const MAX_RECORDS_PER_SHARD = 1000;
+const BYTES_INCOMING_PER_SHARD = 1024 * 1024;
+const BYTES_OUTGOING_PER_SHARD = 2048 * 1024;
+
+const formatFloat = num => {
+	return Number.parseFloat(num).toFixed(3);
+};
+
+class ListKinesisStreamsCommand extends Command {
+	async run() {
+		const { flags } = this.parse(ListKinesisStreamsCommand);
+		const { region, profile } = flags;
+
+		AWS.config.region = region;
+		if (profile) {
+			AWS.config.credentials = new AWS.SharedIniFileCredentials({ profile });
+		}
+
+		checkVersion();
+		let streamsDescription = [];
+		if (region) {
+			this.log(`Checking Kinesis streams in [${region}]`);
+			streamsDescription = await getStreamsDescriptionFromRegion(region);
+		} else {
+			this.log("Checking Kinesis streams in all regions");
+			streamsDescription = await getStreamsDescriptionFromAllRegions();
+		}
+		show(streamsDescription);
+	}
+}
+
+ListKinesisStreamsCommand.description = "Lists the Kinesis streams";
+ListKinesisStreamsCommand.flags = {
+	region: flags.string({
+		char: "r",
+		description: "AWS region, e.g. us-east-1",
+		required: false
+	}),
+	profile: flags.string({
+		char: "p",
+		description: "AWS CLI profile name",
+		required: false
+	})
+};
+
+const getStreamsDescriptionFromRegion = async region => {
+	const Kinesis = new AWS.Kinesis({ region });
+	let streamDetails = await Kinesis.listStreams({ Limit: 100 }).promise();
+	let streamNames = streamDetails.StreamNames;
+	while (streamDetails.HasMoreStreams) {
+		streamDetails = await Kinesis.listStreams({
+			Limit: 100,
+			ExclusiveStartStreamName: streamDetails.slice(-1)[0]
+		}).promise();
+		streamNames = streamDetails.StreamNames;
+	}
+	const usageStats = await getUsageMetrics(streamNames, region);
+	return describeStreams(streamNames, region, usageStats);
+};
+
+const getStreamsDescriptionFromAllRegions = async () => {
+	const promises = Lambda.regions.map(region =>
+		getStreamsDescriptionFromRegion(region)
+	);
+	const results = await Promise.all(promises);
+	return _.flatMap(results);
+};
+
+const describeStreams = async (streams, region, usageStats) => {
+	const streamsDescription = [];
+	for (const stream of streams) {
+		streamsDescription.push(await describeStream(stream, region, usageStats[stream]));
+	}
+
+	return streamsDescription;
+};
+
+const describeStream = async (streamName, region, streamStat) => {
+	const Kinesis = new AWS.Kinesis({ region });
+	const resp = await Kinesis.describeStream({
+		StreamName: streamName
+	}).promise();
+
+	return {
+		streamName: streamName,
+		arn: resp.StreamDescription.StreamARN,
+		status: resp.StreamDescription.StreamStatus,
+		shards: resp.StreamDescription.Shards,
+		readMbPercentage: formatFloat(
+			(streamStat["GetRecords.BytesSum"] /
```

Comment:
this calculation is kinda hard to follow, maybe extract it into a method and use a couple of intermediate variables so one can follow the steps of the calculation easier?

Reply:
Done